### PR TITLE
[CPM-29] Export course plan for student

### PR DESCRIPTION
**Pull request by Richardng12**
## Changes

- Added Export to JSON, Export to MD on plan page

## Checklist

- [x] Have tests been written?
- [x] Is the UI consistent?

## Requests
- [ ] Can somebody check the styling/CSS?

- Ignore plan-route.js changes, my prettier is stuffing things up :(

Addresses [CPM-29]

Author @HongShi10  
Co Auth @Richardng12  @FocalChord 

[CPM-29]: https://momentum-agile.atlassian.net/browse/CPM-29

### CODE REVIEW

**Comment by FocalChord** [comment]
change pr title ty

File: client/src/pages/Plan/Plan.js [hunk 1]
```diff
@@ -10,6 +10,10 @@ import RequirementsList from "./RequirementsList";
 import { colors as c } from "../../colors";
 import { useHotkeys } from "react-hotkeys-hook";
 import { useHistory } from "react-router-dom";
+import json2md from "json2md";
+import { parsePlanWithStudent } from "./JsonCustomConverter";
+import fileDownload from "js-file-download";
+// import mdToPdf from "md-to-pdf";
```
Comment (FocalChord) [inline client/src/pages/Plan/Plan.js:16]: remove commented import

File: client/src/pages/Plan/Plan.js [hunk 2]
```diff
@@ -70,18 +74,20 @@ const Plan = () => {
     };
 
     // TODO: Export Functions
```
Comment (FocalChord) [inline client/src/pages/Plan/Plan.js:76]: is todo done?

File: client/src/pages/Plan/Plan.js [hunk 3]
```diff
@@ -70,18 +74,20 @@ const Plan = () => {
...
-    const exportToMarkdown = (plan) => {
-        console.log("Exporting to Markdown");
+    const exportToMarkdown = () => {
+        const customPlan = parsePlanWithStudent(plan, student, programme, startYear);
+        console.log(json2md(customPlan));
```
Comment (FocalChord) [inline client/src/pages/Plan/Plan.js:79]: remove console.log

File: client/src/pages/Plan/Plan.js [hunk 4]
```diff
@@ -70,18 +74,20 @@ const Plan = () => {
...
+        fileDownload(json2md(customPlan), `${student.name}-${plan.name}.md`);
     };
 
-    const exportToPDF = (plan) => {
+    const exportToJSON = async () => {
+        fileDownload(JSON.stringify(plan), `${student.name}-${plan.name}.json`);
         console.log("Exporting to PDF");
```
Comment (FocalChord) [inline client/src/pages/Plan/Plan.js:85]: remove this console.log too

**Comment by FocalChord** [comment]
Also why are the api files committed when they are formatting changes haha

**Comment by Richardng12** [comment]
> 
> 
> Also why are the api files committed when they are formatting changes haha

i did git add . :(

**Comment by daltonlim** [comment]
lgtm. tho it dies on template page

**Comment by Richardng12** [comment]
> 
> 
> lgtm. tho it dies on template page

F

**Comment by daltonlim** [comment]
> > lgtm. tho it dies on template page
> 
> Can we make it so it doesnt appear on the template page?
> 
> Its cause it down

we could but the only difference is it doesnt have a student. Can you make it so that it checks if student has a value before grabbing its name field

**Comment by Richardng12** [comment]
> 
> 
> > > lgtm. tho it dies on template page
> > 
> > 
> > Can we make it so it doesnt appear on the template page?
> > Its cause it down
> 
> we could but the only difference is it doesnt have a student. Can you make it so that it checks if student has a value before grabbing its name field

do we want them to be able to download the plan from the template pg? I removed it for now, but can add it back in if we need it

**Review by FocalChord — APPROVED** [review]
absolute scenes

**Review by daltonlim — APPROVED** [review]
LGTM

**Comment by ffMathy** [comment]
@pull-dog up

**Pull request merged**